File: src/main/java/ru/fess38/finance/dao/UserDao.java
package ru.fess38.finance.dao;

import org.hibernate.criterion.DetachedCriteria;
import ru.fess38.finance.model.User;

import java.util.List;

public interface UserDao {
  User save(User user);

  User get(long id);

  User update(User user);

  User delete(User user);

  List<User> find(DetachedCriteria detachedCriteria);

  List<User> findDeleted(DetachedCriteria detachedCriteria);

  DetachedCriteria detachedCriteria();
}

File: kotlin/src/main/kotlin/ru/fess38/finance/security/UserDao.kt
package ru.fess38.finance.security

import ru.fess38.finance.core.Model.RefreshToken.AuthType

interface UserDao {
  fun save(user: User)

  fun update(user: User)

  fun find(token: String): User?

  fun find(outerId: String, authType: AuthType): User?
}
